pyfolio/round_trips.py

```diff
@@ -13,168 +13,253 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 from __future__ import division
-from collections import defaultdict
+from math import copysign
+import warnings
+from collections import deque, OrderedDict
 
 import pandas as pd
 import numpy as np
 
-
-def extract_round_trips(transactions):
-    """
-    Group transactions into "round trips." A round trip is started when a new
-    long or short position is opened and is only completed when the number
-    of shares in that position returns to or crosses zero.
-    Computes pnl for each round trip.
-
-    For example, the following transactions would constitute one round trip:
-    index                  amount   price    symbol
-    2004-01-09 12:18:01    186      324.12   'AAPL'
-    2004-01-09 15:12:53    -10      344.54   'AAPL'
-    2004-01-13 14:41:23    24       320.21   'AAPL'
-    2004-01-30 10:23:34    -200     340.43   'AAPL'
+from .utils import print_table
+
+PNL_STATS = OrderedDict(
+    [('Total profit', lambda x: x.sum()),
+     ('Gross profit', lambda x: x[x > 0].sum()),
+     ('Gross loss', lambda x: x[x < 0].sum()),
+     ('Profit factor', lambda x: x[x > 0].sum() / x[x < 0].abs().sum()
+      if x[x < 0].abs().sum() != 0 else np.nan),
+     ('Avg. trade net profit', 'mean'),
+     ('Avg. winning trade', lambda x: x[x > 0].mean()),
+     ('Avg. losing trade', lambda x: x[x < 0].mean()),
+     ('Ratio Avg. Win:Avg. Loss', lambda x: x[x > 0].mean() /
+      x[x < 0].abs().mean() if x[x < 0].abs().mean() != 0 else np.nan),
+     ('Largest winning trade', 'max'),
+     ('Largest losing trade', 'min'),
+     ])
+
+SUMMARY_STATS = OrderedDict(
+    [('Total number of round_trips', 'count'),
+     ('Percent profitable', lambda x: len(x[x > 0]) / float(len(x))),
+     ('Winning round_trips', lambda x: len(x[x > 0])),
+     ('Losing round_trips', lambda x: len(x[x < 0])),
+     ('Even round_trips', lambda x: len(x[x == 0])),
+     ])
+
+RETURN_STATS = OrderedDict(
+    [('Avg returns all round_trips', lambda x: x.mean()),
+     ('Avg returns winning', lambda x: x[x > 0].mean()),
+     ('Avg returns losing', lambda x: x[x < 0].mean()),
+     ('Median returns all round_trips', lambda x: x.median()),
+     ('Median returns winning', lambda x: x[x > 0].median()),
+     ('Median returns losing', lambda x: x[x < 0].median()),
+     ('Largest winning trade', 'max'),
+     ('Largest losing trade', 'min'),
+     ])
+
+DURATION_STATS = OrderedDict(
+    [('Avg duration', lambda x: x.mean()),
+     ('Median duration', lambda x: x.median()),
+     ('Avg # round_trips per day', lambda x: float(len(x)) /
+      (x.max() - x.min()).days),
+     ('Avg # round_trips per month', lambda x: float(len(x)) /
+      (((x.max() - x.min()).days) / 21)),
+     ])
+
+
+def agg_all_long_short(round_trips, col, stats_dict):
+    stats_all = round_trips.assign(ones=1)\
```

Comment:
Most examples of method chaining I've seen put parenthesis around the whole expression instead of using `\` 
ex: http://pandas.pydata.org/pandas-docs/stable/basics.html#tablewise-function-application

Not a big deal though.